sdk-base/src/main/java/com/mapbox/maps/extension/observable/ObservableExtension.kt

```diff
@@ -5,8 +5,12 @@ import com.mapbox.maps.Event
 import com.mapbox.maps.MapEvents
 import com.mapbox.maps.ObservableInterface
 import com.mapbox.maps.Observer
-import com.mapbox.maps.extension.observable.model.*
-import com.mapbox.maps.extension.observable.resourcerequest.ResourceEventData
+import com.mapbox.maps.extension.observable.eventdata.*
+import com.mapbox.maps.extension.observable.eventdata.ResourceEventData
+
+private val gson by lazy {
+  Gson()
```

Comment:
however could it technically happen that user is parsing several instances of event data from multiple threads? guess it's too advanced though 😄 so let's keep as is

Reply:
According to Gson 's [api doc](https://www.javadoc.io/doc/com.google.code.gson/gson/2.8.0/com/google/gson/Gson.html)

```
Gson is typically used by first constructing a Gson instance and then invoking toJson(Object) or fromJson(String, Class) methods on it. Gson instances are Thread-safe so you can reuse them freely across multiple threads.
```

Reply:
how about [koltin serialization](https://github.com/Kotlin/kotlinx.serialization)?
maybe not in this PR

Reply:
> how about [koltin serialization](https://github.com/Kotlin/kotlinx.serialization)? maybe not in this PR

we were using gson library to serialise/deserialise in our codebase, it would be good to evaluate and replace it across the codebase in a separate PR.